internal/action/setup.go

```diff
@@ -0,0 +1,287 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/fatih/color"
+	"github.com/gopasspw/gopass/internal/backend"
+	"github.com/gopasspw/gopass/internal/backend/crypto/gpg"
+	"github.com/gopasspw/gopass/internal/out"
+	"github.com/gopasspw/gopass/internal/store/root"
+	"github.com/gopasspw/gopass/pkg/ctxutil"
+	"github.com/gopasspw/gopass/pkg/debug"
+	"github.com/gopasspw/gopass/pkg/pwgen/xkcdgen"
+	"github.com/gopasspw/gopass/pkg/termio"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+// Setup will invoke the onboarding / setup wizard
+func (s *Action) Setup(c *cli.Context) error {
+	ctx := ctxutil.WithGlobalFlags(c)
+	remote := c.String("remote")
+	team := c.String("alias")
+	create := c.Bool("create")
+
+	ctx = initParseContext(ctx, c)
+
+	out.Print(ctx, logo)
+	out.Print(ctx, "🌟 Welcome to gopass!")
+	out.Print(ctx, "🌟 Initializing a new password store ...")
+
+	if name := termio.DetectName(c.Context, c); name != "" {
+		ctx = ctxutil.WithUsername(ctx, name)
+	}
+	if email := termio.DetectEmail(c.Context, c); email != "" {
+		ctx = ctxutil.WithEmail(ctx, email)
+	}
+	// need to re-initialize the root store or it's already initialized
+	// and won't properly set up crypto according to our context.
+	s.Store = root.New(s.cfg)
+	inited, err := s.Store.IsInitialized(ctx)
+	if err != nil {
+		return ExitError(ExitUnknown, err, "Failed to initialized store: %s", err)
+	}
+	if inited {
+		out.Error(ctx, "⚠ Store is already initialized. Aborting.")
+		return nil
+	}
+
+	debug.Log("Starting Onboarding Wizard - remote: %s - team: %s - create: %t - name: %s - email: %s", remote, team, create, ctxutil.GetUsername(ctx), ctxutil.GetEmail(ctx))
+
+	crypto := s.getCryptoFor(ctx, team)
+	if crypto == nil {
+		return fmt.Errorf("can not continue without crypto")
+	}
+	debug.Log("Crypto Backend initialized as: %s", crypto.Name())
+
+	// check for existing GPG keypairs (private/secret keys). We need at least
+	// one useable key pair. If none exists try to create one
+	if !s.initHasUseablePrivateKeys(ctx, crypto) {
+		out.Print(ctx, "🔐 No useable cryptographic keys. Generating new key pair")
+		out.Print(ctx, "🕰 Key generation may take up to a few minutes")
+		if err := s.initGenerateIdentity(ctx, crypto, ctxutil.GetUsername(ctx), ctxutil.GetEmail(ctx)); err != nil {
+			return errors.Wrapf(err, "failed to create new private key")
+		}
+		out.Print(ctx, "🔐 Cryptographic keys validated")
```

Comment:
Are they "validated" or "generated" at this point?

Reply:
generated seems more apt.